includes/Plugin_Context.php

```diff
@@ -42,7 +43,13 @@ class Plugin_Context {
 	 * @param string $main_file The absolute path to the plugin main file.
 	 */
 	public function __construct( $main_file ) {
-		$this->main_file = wp_normalize_path( $main_file );
+		if ( function_exists( '\WP_CLI\Utils\normalize_path' ) ) {
+			$this->main_file = normalize_path( $main_file );
+		} elseif ( function_exists( 'wp_normalize_path' ) ) {
+			$this->main_file = wp_normalize_path( $main_file );
+		} else {
+			$this->main_file = $main_file;
```

Comment:
Following up on my previous feedback, I do think it's better to throw an exception or at least trigger a PHP warning here because it will result in problems when using Windows if this logic is ever reached.

It should never be reached in the first place, which is why I think even an exception here is fine. But not doing anything and silently let it pass will lead to obscure errors, should anyone ever run into it.